Source/Applications/SystemCenter/Controllers/OpenXDA/Meters/OpenXDAMeterController.cs

```diff
@@ -486,6 +486,36 @@ public IHttpActionResult DeleteMeterAsset(int meterID, int assetID, int location
             
         }
 
+        [HttpPost, Route("{meterID:int}/Asset/{assetID:int}/{locationID:int}")]
+        public IHttpActionResult AddMeterAsset(int meterID, int assetID, int locationID)
+        {
+            if (PostRoles == string.Empty || User.IsInRole(PostRoles))
+            {
+                using (AdoDataConnection connection = new AdoDataConnection(Connection))
+                {
+                    try
+                    {
+                        MeterAsset meterAsset = new TableOperations<MeterAsset>(connection).QueryRecordWhere("MeterID = {0} AND AssetID = {1}", meterID, assetID);
+                        if (meterAsset is null) 
+                            new TableOperations<MeterAsset>(connection).AddNewRecord(new MeterAsset() { MeterID = meterID, AssetID = assetID});
+
+                        AssetLocation assetLoc = new TableOperations<AssetLocation>(connection).QueryRecordWhere("LocationID = {0} AND AssetID = {1}", locationID, assetID);
+                        if (assetLoc is null)
+                            new TableOperations<AssetLocation>(connection).AddNewRecord(new AssetLocation() { LocationID = locationID, AssetID = assetID });
+                     
+                        return Ok(1);
+                    }
+                    catch (Exception ex)
+                    {
+                        return InternalServerError(ex);
+                    }
```

Comment:
Does this make any difference compared to simply not having a try-catch block at all?

Reply:
the InternalServerError(ex) guarantees a type we can use on the client side. What I don't know is if the response would be the same if we throw an exception (it may be wrapping the Controller endpoint in a try-catch internally and return a InternalServerError object regardless.

Reply:
Yeah, that's exactly what the web server does. It's why exceptions always result in 500 Internal Server Error and we have to try to use the developer console to get the exception message. AFAIK, the only thing this could potentially change is if we ever use this function directly like in the following snippet.

```c#
var controller = new OpenXDAMeterController();
var result = controller.AddMeterAsset(meterID, assetID, locationID);
```

I have seen this kind of thing in our code, but this is extremely unlikely to be used in any case except one controller proxying another controller, which is another case in which the try-catch doesn't matter.

Reply:
It shouldn't matter if it's in IIS or self-hosted--they work the same in this respect. The downside here is similar to the downside of writing code like `if ((days >= 30) == true ? true : false)`. It will make programmers stop and wonder why it's there. Defensive programming habits will cause it to be proliferated into every method in lieu of proper exception handling, which I feel is what's going on here already.

Perhaps more importantly, if you don't let the exception bubble up, then the caller will have no opportunity to handle the exception properly. For instance, this type of code would make it harder to implement a global exception handler for logging these types of errors.

https://learn.microsoft.com/en-us/aspnet/web-api/overview/error-handling/web-api-global-error-handling